homeassistant/components/timer/__init__.py

```diff
@@ -185,6 +214,15 @@ def __init__(self, config: typing.Dict):
         self.editable = True
         self._state = STATUS_IDLE
         self._remaining = config[CONF_DURATION]
+        self._restore = config[CONF_RESTORE] if config[CONF_RESTORE] is not None \
```

Comment:
This doesn't work. `config[CONF_RESTORE]` raises a `KeyError` because the data is not in the config.

Reply:
You should replace this line with `config.get(CONF_RESTORE, DEFAULT_RESTORE)`